### PR TITLE
feat(test-runner): use require.resolve for globalSetup and globalTeardown

### PR DESCRIPTION
**Pull request by JoelEinbinder**
This patch is done as to not be a breaking change.

### CODE REVIEW

File: src/test/loader.ts [hunk 1]
```diff
@@ -437,3 +438,10 @@ const baseFullConfig: FullConfig = {
   workers: 1,
   launch: [],
 };
+
+function resolveScript(id: string, rootDir: string) {
+  const localPath = path.resolve(rootDir, id);
+  if (fs.existsSync(localPath))
+    return localPath;
+  return require.resolve(id, { paths: [rootDir] });
```
Comment (dgozman) [inline src/test/loader.ts:446]: So, this is going to work for `globalSetup: 'playwright-dev-server'` where you use a node module as a global setup?

File: tests/playwright-test/global-setup.spec.ts [hunk 2]
```diff
@@ -21,7 +21,7 @@ test('globalSetup and globalTeardown should work', async ({ runInlineTest }) =>
     'playwright.config.ts': `
       import * as path from 'path';
       module.exports = {
-        globalSetup: 'globalSetup.ts',
+        globalSetup: './globalSetup',
```
Comment (dgozman) [inline tests/playwright-test/global-setup.spec.ts:24]: Since we preserve the old mode, let's leave a test that uses it without a relative path.

**Review by dgozman — APPROVED** [review]

File: tests/playwright-test/global-setup.spec.ts (hunk 2 above)
Comment (JoelEinbinder) [reply tests/playwright-test/global-setup.spec.ts:24]: there are some down below that i left unchanged.

File: src/test/loader.ts (hunk 1 above)
Comment (JoelEinbinder) [reply src/test/loader.ts:446]: yes

Comment (JoelEinbinder) [reply src/test/loader.ts:453]: added a test to confirm

**Pull request merged**